### ISSUE TITLE
Format of numeric UDAs in report changed

### ISSUE THREAD

**Issue by jkaiser-ioki** [GothenburgBitFactory/taskwarrior#3306]
**Issue appeared with upgrade to taskwarrior 3.0.0**

I have a custom UDA defined like this:
```
uda.size.type=numeric
uda.size.label=Size
uda.size.values=1,2,3,5,8,13,21
```
I added the column to my `next` report:
```
report.next.labels=ID,Active,Age,Deps,P,Project,Tag,Recur,Sched,Due,Until,Size,Description,Urg
report.next.columns=id,start.age,entry.age,depends,priority,project,tags,recur,scheduled.countdown,due.relative,until.remaining,size,description,urgency
```

When I run `task` to get the `next` report, the "Size" column shows up as expected, but its values are now displayed as floating point numbers with six digits after the decimal point, rather than as integers:
![image](https://github.com/GothenburgBitFactory/taskwarrior/assets/46380487/b29d1eb0-1d43-43eb-946e-e71ff3544a41)

This is not critical, but widens the rendering of the table unnecessarily. I guess it would be nice if it were either possible to declare the UDA type more specifically as int vs float, or to at least have a possibility to format the rendering of a particular column somehow.

I guess the behaviour is due to the changed backend somehow and the internal representation of the values

**Comment by djmitche** [comment]
I suspect this is due to the export/import process -- can you check whether newly-added tasks still have the zeroes for this field?

**Comment by jkaiser-ioki** [comment]
good guess, here's a newly created task:

![image](https://github.com/GothenburgBitFactory/taskwarrior/assets/46380487/777bf8b9-84a1-44c6-b253-0ff247b32047)

and using something like `task 116 mod size:5` also fixes that particular task ...

question: is there a simple solution that modfies all tasks in one batch run ...

**Comment by jkaiser-ioki** [comment]
ok, doing something like this
```
[jonas@orchid ~]$ task size:8 mod size:8
This command will alter 111 tasks.
  - Size will be changed from '8.000000' to '8'.
```

for all the allowed values in size seems to do the trick (fortunately it is only a handful of possible values :sweat_smile: )

**Comment by djmitche** [comment]
Nice! Any chance you could include an edited-down, redacted version of a line or two of your `task export` output?

**Comment by jkaiser-ioki** [comment]
> Nice! Any chance you could include an edited-down, redacted version of a line or two of your `task export` output?

I don't believe that will be possible, since I deleted the exported file after "successful" import (and it wasn't captured by by backup routine either :see_no_evil: ).  And also the original storage files in the .task directory (as suggested by the migration instructions)

**Comment by djmitche** [comment]
Bummer!

3.0 didn't really change the `task import` / `task export` implementations, but it has certainly encouraged more people to run them, and I think we're finding some of the "edge cases" where values don't survive the round trip. One thing I didn't realize is that the format of the export depends on the type of UDAs, so a numeric UDA is exported as a JSON number, despite being stored as a string internally to Taskwarrior.

Based on that information, maybe someone can cook up a reproduction recipe for this?

**Comment by djmitche** [comment]
Forgot to add: I'm not prioritizing this because I think the bug here is in the `export`, so a fix in 3.0 won't actually help users transitioning to 3.0. We should still fix the bug though, and a repro recipe will help!

**Comment by GhostOrder28** [comment]
Hello, I'm also experiencing this bug in existing tasks, however in my case doing: `task point:8 mod point:8` didn't fix it, manually editing the task (task `id` edit) also doesn't fix it:

my config:
uda.point.type=numeric
uda.point.label=Point

Any clues?

PD: I'm in version 3.4.1

**Comment by djmitche** [comment]
Can you show how to reproduce this from an empty task database?

**Comment by apoelstra** [comment]
FYI I believed I had this bug, but after attempting to reproduce I realized that it was my on-add hook that was doing it. The bug eventually came down to [tasklib::prelude::UdaValue](https://docs.rs/tasklib/latest/tasklib/prelude/enum.UdaValue.html) always storing UDA numerics as floats, so when they are reserialized in my hook logic, a bunch of extra .0s get added.

**Comment by djmitche** [comment]
Thanks for the update!

**Comment by apoelstra** [comment]
Oops -- I'm not the OP here and maybe the issue should remain open.

**Comment by djmitche** [comment]
I did miss that you weren't the OP -- but the OP hasn't updated the issue in a while and it's still un-reproducible, so I think closed is a good state.
